response_rewriter_test.go

```diff
@@ -295,6 +295,55 @@ func TestIsMasterResponseRewriterSuccess(t *testing.T) {
 		"primary": "2",
 		"foo":     "bar",
 	}
+
+	r := &IsMasterResponseRewriter{
+		Log:                 nopLogger{},
+		ProxyMapper:         proxyMapper,
+		ReplicaStateCompare: fakeReplicaStateCompare{sameIM: true, sameRS: true},
+		ReplyRW: &ReplyRW{
+			Log: nopLogger{},
+		},
+	}
+
+	var client bytes.Buffer
+	if err := r.Rewrite(&client, fakeSingleDocReply(in)); err != nil {
+		t.Fatal(err)
+	}
+	actualOut := bson.M{}
+	doc := client.Bytes()[headerLen+len(emptyPrefix):]
+	if err := bson.Unmarshal(doc, &actualOut); err != nil {
+		t.Fatal(err)
+	}
+	if !reflect.DeepEqual(out, actualOut) {
+		spew.Dump(out)
+		spew.Dump(actualOut)
+		t.Fatal("did not get expected output")
+	}
+}
+
+func TestIsMasterResponseRewriterSuccessWithPassives(t *testing.T) {
```

Comment:
can we add a test about that error case https://github.com/intercom/dvara/pull/21/files#diff-097318650f3ace0d9bd6b33a97155a57R383 ?